src/service/database/databaseservice.ts

```diff
@@ -150,10 +173,19 @@ export class DatabaseService extends RestService {
 	 * @return a JSON object document result
 	 */
 	public async findOne(collection: string, filter: JSONObject, options?: FindOptions): Promise<JSONObject> {
-		let docs = await this.findMany(collection, filter, options);
+        let efilter = filter;
+        if (this.encrypt) {
+            efilter = this.databaseEncrypt.encryptFilter(filter);
+        }
 
-		DatabaseService.LOG.debug(JSON.stringify(docs));
-		return docs !== undefined && !(docs.length === 0) ? docs[0] : null;
+		let docs = await this.findMany(collection, efilter, options);
+
+		DatabaseService.LOG.debug(`fine docs: ${JSON.stringify(docs)}`);
+		if (!docs || docs.length === 0) {
```

Comment:
Should we throw exception here, or catch exception from line 181.

Reply:
Method findMany() is an internal call here.